components/automate-cli/cmd/chef-automate/upgrade.go

```diff
@@ -117,9 +120,27 @@ func runUpgradeCmd(cmd *cobra.Command, args []string) error {
 		writer.Println("Chef Automate up-to-date")
 		return nil
 	}
+	var ReadPendingPostChecklist = []string{}
+	_, is_major_version := manifest.IsSemVersionFmt(validatedResp.CurrentVersion)
+	if is_major_version {
+		var err error
+		ci, err := majorupgradechecklist.NewPostChecklistManager(validatedResp.CurrentVersion)
+		if err != nil {
+			return err
+		}
+
+		ReadPendingPostChecklist, err = ci.ReadPendingPostChecklistFile()
+		if err != nil {
+			return status.Wrap(
+				err,
+				status.DeploymentServiceCallError,
+				"unable to read checklist file",
+			)
+		}
+	}
```

Comment:
I think we can put all this in a function which returns ([]string, error) 
```go
func GetPendingPostChecklist() ([]string, error) {
        _, is_major_version := manifest.IsSemVersionFmt(validatedResp.CurrentVersion)
	if is_major_version {
		var err error
		ci, err := majorupgradechecklist.NewPostChecklistManager(validatedResp.CurrentVersion)
		if err != nil {
			return []string{}, err
		}

		pendingPostChecklist, err = ci.ReadPendingPostChecklistFile()
		if err != nil {
			return []string{}, fmt.Errorf("unable to read checklist file: %s", err)
		}
                return pendingPostChecklist, nil
        }
        return []string{}, nil
}
```
and Use this as this here:
```go
pendingPostChecklist, err := GetPendingPostChecklist()
if err != nil {
        return status.Wrap(err, status.DeploymentServiceCallError, "unable to read checklist file")
}
```